lucene/misc/src/java/org/apache/lucene/misc/store/DirectIODirectory.java

```diff
@@ -154,14 +146,43 @@ public IndexOutput createOutput(String name, IOContext context) throws IOExcepti
     if (context.context != Context.MERGE || context.mergeInfo.estimatedMergeBytes < minBytesDirect) {
       return delegate.createOutput(name, context);
     } else {
-      return new NativeUnixIndexOutput(getDirectory().resolve(name), name, mergeBufferSize);
+      return new DirectIOIndexOutput(getDirectory().resolve(name), name, mergeBufferSize);
     }
   }
 
-  @SuppressForbidden(reason = "java.io.File: native API requires old-style FileDescriptor")
-  private final static class NativeUnixIndexOutput extends IndexOutput {
+  @Override
+  public void deleteFile(String name) throws IOException {
+    delegate.deleteFile(name);
+  }
+
+  @Override
+  public Set<String> getPendingDeletions() throws IOException {
+    return delegate.getPendingDeletions();
+  }
+
+  @Override
+  public String[] listAll() throws IOException {
+    return delegate.listAll();
+  }
+
+  @Override
+  public long fileLength(String name) throws IOException {
+    return delegate.fileLength(name);
+  }
+
+  @Override
+  public void rename(String source, String dest) throws IOException {
+    delegate.rename(source, dest);
+  }
+
+  @Override
+  public void close() throws IOException {
+    delegate.close();
```

Comment:
Maybe use `IOUtils.close(...)` here instead?  It will try to close both even if one of them throws exception.

Reply:
I made the update to replace `delegate.close()` with `IOUtils.close(delegate)`, but I don't think I could achieve `super.close()` through `IOUtils.close(this)` as well, as that leads to stackoverflow? Or am I misinterpreting the comment here?

Reply:
Oh yeah hmm, that is frustrating :)  I was hoping we could pass two things to `IOUtils.close` ... maybe pass `FilterDirectory.this`?  Though I suspect that would indeed just call this existing `close()` method, leading to infinite recursion.  Hrmph, this is beyond my Java knowledge but sure seems like it ought to be possible.  Maybe @uschindler knows :)